src/Microsoft.AspNet.Mvc.ModelBinding/BinderMetadata/CompositeBindingSource.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A <see cref="BindingSources"/> which can repesent multiple value-provider data sources.
+    /// </summary>
+    public class CompositeBindingSource : BindingSource
```

Comment:
This is really only needed by the framework for representing `[FromUrl]` (webapi). I felt like this was a much better approach than making the interface return `IEnumerable<BindingSource>` and pushing the complexity of dealing with multiples throughout all of this code.

The result of this was that basically no changes to model-binder logic were needed for this swap.